pkg/server/routes.go

```diff
@@ -33,24 +34,40 @@ import (
 	"github.com/cmsgov/easi-app/pkg/handlers"
 	"github.com/cmsgov/easi-app/pkg/local"
 	"github.com/cmsgov/easi-app/pkg/models"
+	"github.com/cmsgov/easi-app/pkg/okta"
 	"github.com/cmsgov/easi-app/pkg/services"
 	"github.com/cmsgov/easi-app/pkg/storage"
 	"github.com/cmsgov/easi-app/pkg/upload"
 )
 
 func (s *Server) routes(
-	authorizationMiddleware func(handler http.Handler) http.Handler,
 	corsMiddleware func(handler http.Handler) http.Handler,
 	traceMiddleware func(handler http.Handler) http.Handler,
 	loggerMiddleware func(handler http.Handler) http.Handler) {
 
+	// TODO: We should add some sort of config verifier to make sure these configs exist
+	// They may live in /cmd, but should fail quick on startup
+	oktaAuthenticationMiddleware := okta.NewOktaAuthenticationMiddleware(
+		handlers.NewHandlerBase(s.logger),
+		s.Config.GetString("OKTA_CLIENT_ID"),
+		s.Config.GetString("OKTA_ISSUER"),
+		s.Config.GetBool("ALT_JOB_CODES"),
+	)
+
 	s.router.Use(
 		traceMiddleware, // trace all requests with an ID
 		loggerMiddleware,
 		corsMiddleware,
-		// authorizationMiddleware, // is supposed to be authN, not authZ; TODO: those responsibilities should be split out
+		oktaAuthenticationMiddleware,
 	)
 
+	if s.Config.GetString(appconfig.LocalAuth) == "enabled" {
+		localAuthenticationMiddleware := local.NewLocalAuthenticationMiddleware(s.logger, s.Config.GetString(appconfig.LocalAuth))
+		s.router.Use(localAuthenticationMiddleware)
```

Comment:
Does this replace the okta middleware? Or just add to it? Can I run the app locally if I can't connect to Okta?

Reply:
It adds it to the stack. Both can run on any request now; they just set the principal on the context so that other parts of the app (say, the `require_principal_middleware`) can determine if a user is logged in.

The driver of this design is that we want to be able to run tests on CI that use local auth OR okra, and so we'll need to be able to log in with either one.